### PR TITLE
docs: Azure Disk Encryption (PCP-3720 & PCP-3735)

### PR DESCRIPTION
**Pull request by benradstone**
## Describe the Change

Documentation for the new Azure Disk Encryption support using Disk Encryption Sets with customer-managed keys.

Only available to Azure IaaS for 4.5.a.

## Changed Pages

💻 [Azure Disk Encryption](https://deploy-preview-4558--docs-spectrocloud.netlify.app/clusters/public-cloud/azure/azure-disk-encryption/)

## Jira Tickets

🎫 [PCP-3720](https://spectrocloud.atlassian.net/browse/PCP-3720)
🎫 [PCP-3735](https://spectrocloud.atlassian.net/browse/PCP-3735)

## Backports

Can this PR be backported?

- [ ] Yes. _Remember to add the relevant backport labels to your PR._
- [x] No. _Please leave a short comment below about why this PR cannot be backported._

4.5.a release ticket.

[PCP-3720]: https://spectrocloud.atlassian.net/browse/PCP-3720?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PCP-3735]: https://spectrocloud.atlassian.net/browse/PCP-3735?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+
+   1. Log in to the [Azure Portal](https://portal.azure.com/).
+
+   2. In the search bar, look for **Disk Encryption Sets**. Click on the service when found.
+
+   3. Find your Disk Encryption Set from the list and click on it to view details.
+
+   4. On the **Overview** page, click **JSON View** in the **Essentials** section. The Resource ID for the Disk
+      Encryption Set is displayed at the top.
+   5. Click the **Copy to clipboard** icon for the Resource ID and paste it into the `diskEncryptionSetID` field in the
```
Comment (caroldelwing) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:206]: we should also tell users to replace `<subscriptionId>` and `<resourceGroup>` with their information

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+   cloud:
+     azure:
+       diskEncryptionSetID: "/subscriptions/<subscriptionId>/resourceGroups/<resourceGroup>/providers/Microsoft.Compute/diskEncryptionSets/<diskEncryptionSet>"
+   ```
+
+10. Click **Confirm updates**.
+
+11. Click **Save Changes**.
+
+#### Enable Disk Encryption on an Active Cluster
```
Comment (caroldelwing) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:221]: I think we could merge the  `Enable Disk Encryption on a Cluster Profile` and `Enable Disk Encryption on an Active Cluster` sub guides, once one is a continuation of the other. Then, we could have only one h3 heading `Enable Disk Encryption on an Active Cluster` . What do you think?

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+Use the following steps to disable disk encryption on an
+[existing cluster profile](../../../profiles/cluster-profiles/modify-cluster-profiles/modify-cluster-profiles.md).
+
+1. Log in to [Palette](https://console.spectrocloud.com).
+
+2. Ensure you are in the correct project scope.
+
+3. From the left **Main Menu**, select **Profiles** and click the cluster profile that you want to edit.
+
+4. Create a new version of your cluster profile:
```
Comment (caroldelwing) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:262]: Same comment about sup-steps

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+6. Review the changes and click **Update**.
+
+Your cluster will now update and a full cluster repave will occur. Wait until the update has completed before
+[validating the disk encryption disablement](#validate-after-disablement).
+
+## Validate
+
+### Validate after Enablement
+
+Follow these steps to validate enablement of customer-managed key encryption on your Azure VM disks.
```
Comment (caroldelwing) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:311]:
```suggestion
Follow these steps to validate the enablement of customer-managed key encryption on your Azure VM disks.
```

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+4. Once identified, click on a disk name to view its details.
+
+5. Scroll down to view the **Properties** tab and check that the **Encryption** section shows **Customer-managed key**
+   for **Encryption type**. Your Disk Encryption Set name is also shown for the **Encryption key**.
+
+6. Repeat steps 4-5 for each disk in your cluster.
+
+### Validate after Disablement
+
+Follow these steps to validate disablement of customer-managed key encryption on your Azure VM disks.
```
Comment (caroldelwing) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:329]:
```suggestion
Follow these steps to validate the disablement of customer-managed key encryption on your Azure VM disks.
```

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+
+3. Find your cluster's disks by using the search filters provided. You can use the cluster name as the cluster resources
+   contain the cluster name at the beginning (for example: `<clusterName>-e3c0-f7ljd_OSDisk`).
+
+4. Once identified, click on a disk name to view its details.
+
+5. Scroll down to view the **Properties** tab and check that the **Encryption** section shows **Platform-managed key**
+   for **Encryption type**.
+
+6. Repeat steps 4-5 for each disk in your cluster.
```
Comment (caroldelwing) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:343]:
```suggestion
6. Repeat steps four and five for each disk in your cluster.
```

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+
+3. Find your cluster's disks by using the search filters provided. You can use the cluster name as the cluster resources
+   contain the cluster name at the beginning (for example: `<clusterName>-e3c0-f7ljd_OSDisk`).
+
+4. Once identified, click on a disk name to view its details.
+
+5. Scroll down to view the **Properties** tab and check that the **Encryption** section shows **Customer-managed key**
+   for **Encryption type**. Your Disk Encryption Set name is also shown for the **Encryption key**.
+
+6. Repeat steps 4-5 for each disk in your cluster.
```
Comment (caroldelwing) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:325]:
```suggestion
6. Repeat steps four and five for each disk in your cluster.
```

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+
+  - Key Management Operations:
+
+    - :white_check_mark: **Get**
+
+  - Cryptographic Operations:
+
+    - :white_check_mark: **Unwrap Key**
+    - :white_check_mark: **Wrap Key**
+
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:78]: @benradstone I'm not sure what this is, but it's not our style. The emojis are primarily only reserved for markdown tables with a small set of exceptions.  Consider using a table to list out the required permissions for each resource, or stick with a list. Here is an [example](https://docs.spectrocloud.com/clusters/public-cloud/aws/enable-secrets-encryption-kms-key/) where IAM permissions are specified for a how to.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+1. Log in to the [Azure Portal](https://portal.azure.com/).
+
+2. In the search bar, look for **Disks**. Click on the service when found.
+
+3. Find your cluster's disks by using the search filters provided. You can use the cluster name as the cluster resources
+   contain the cluster name at the beginning (for example: `<clusterName>-e3c0-f7ljd_OSDisk`).
+
+4. Once identified, click on a disk name to view its details.
+
+5. Scroll down to view the **Properties** tab and check that the **Encryption** section shows **Customer-managed key**
```
Comment (caroldelwing) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:322]: Maybe we could merge the `Validate after Enablement` and `Validate after Disablement` guides as well to simplify? Step 5 is the only different step

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+
+5. In **Cloud Type**, under **Infrastructure provider**, select **Azure IaaS**. Click **Next**.
+
+6. Select your base OS pack and configure it to your liking. Select **Next layer** when complete.
+
+7. Select the **Palette eXtended Kubernetes** pack as your base Kubernetes pack.
+
+8. In **Configure Pack**, select **Values** in pack details, and click the **\</\>** button to show the YAML editor.
+
+9. On the right-hand side, click the **Presets** drop-down menu, and select the **Enable Encryption Using
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:105]:
```suggestion
9. On the right-hand side, click the **Presets** drop-down Menu, and select the **Enable Encryption Using
```

File: docs/docs-content/clusters/public-cloud/azure/create-azure-cluster.md [hunk 11]
```diff
@@ -67,6 +67,14 @@ to create an IaaS Kubernetes cluster in Azure that Palette manages.
   about Private DNS Zones, refer to
   [Private DNS Zones](https://learn.microsoft.com/en-us/azure/private-link/private-endpoint-dns).
 
+- If you want to enable Azure Disk Encryption on your cluster, ensure you have created a Key Vault and Disk Encryption
+  Set. Your cluster profile must also be configured to use the
```
Comment (caroldelwing) [inline docs/docs-content/clusters/public-cloud/azure/create-azure-cluster.md:71]: Prettier here again :)

**Review by caroldelwing — CHANGES_REQUESTED** [review]
Excellent work, Ben! :clap: I requested a few minor changes. We can discuss them further if you'd like.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 12]
```diff
@@ -0,0 +1,353 @@
...
+
+11. Fill in the `diskEncryptionSetID` with the Resource ID URI of your Disk Encryption Set.
+
+    <!-- prettier-ignore -->
+    <details>
+    <summary> How to find the Resource ID URI of your Disk Encryption Set </summary>
+
+    1. Log in to the [Azure Portal](https://portal.azure.com/).
+
+    2. In the search bar, look for **Disk Encryption Sets**. Click on the service when found.
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:124]: Let's use more explicit language so that the user knows to click on the search bar and type out the Disk query.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 13]
```diff
@@ -0,0 +1,353 @@
...
+   2. Fill the **Version** field with a new version number.
+   3. Click **Confirm**.
+
+5. Select the Kubernetes layer to view the **Edit Pack** drawer.
+
+   :::warning
+
+   Ensure that you are using
+
+   <VersionedLink text="Palette eXtended Kubernetes (PXK)" url="/integrations/packs/?pack=kubernetes" /> as the Azure
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:174]: Same feedback as earlier, use the prettier ignore to fix the newline injected by prettier.
But, also, this can be addressed by making it a prereq, so we could do without an admonition.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 14]
```diff
@@ -0,0 +1,353 @@
...
+    ```
+
+12. Make any other changes that you need and click **Next layer**.
+
+13. Select the remaining profile layers to finish the configuration.
+
+You can now [create a new Azure IaaS cluster](./create-azure-cluster.md) with disk encryption enabled using this cluster
+profile.
+
+### Edit an Existing Cluster Profile
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:149]: Every how-to needs a prerequisites section and a validation section.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:1]: @benradstone I think there is some confusion about the required sections for how-tos. Each how-to, requires its own prereq, enablement (steps) section, and validation. In this case, you have have multiple how-tos in a single page, which is fine as it's all related to the same topic. But we still need to honor the layout guide.  Here is an [example](https://docs.spectrocloud.com/clusters/cluster-management/backup-restore/add-backup-location-static/) of a guide with multiple how-tos.

TLDR: Treat each how-to as it's own section. Trying to combine a validation and prereq for multiple topics gets confusing from a reader perspective. 


Another minor thing I caught a few times was incorrect UI element labeling. Take a peek at the Product UI Naming table of the[ style guide](https://spectrocloud.atlassian.net/wiki/spaces/DE/pages/1765933057/Spectro+Cloud+Internal+Style+Guide).

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 15]
```diff
@@ -0,0 +1,353 @@
...
+[platform-managed keys](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption#platform-managed-keys),
+however, customer-managed keys enable you to have greater control over your key management.
+
+You can use Disk Encryption Sets to encrypt your nodes' Operating System (OS) and data disks by including a preset
+option in the <VersionedLink text="Palette eXtended Kubernetes (PXK)" url="/integrations/packs/?pack=kubernetes" />
+pack.
+
+## Limitations
+
+- Azure Disk Encryption is available from [4.5.a](../../../release-notes/release-notes.md) onwards.
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:25]: You don't need to do this as our docs are versioned. This content will only be available in new versions of our docs. 

```suggestion
```

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 16]
```diff
@@ -0,0 +1,353 @@
...
+
+- Azure Disk Encryption is only supported on Azure IaaS clusters.
+
+- Azure Disk Encryption is only supported when using the
+
+  <VersionedLink text="Palette eXtended Kubernetes (PXK)" url="/integrations/packs/?pack=kubernetes" /> pack.
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:31]: Fix prettier newline

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 17]
```diff
@@ -0,0 +1,353 @@
...
+    [**Auto key rotation**](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption#automatic-key-rotation-of-customer-managed-keys)
+    on your Disk Encryption Set so it can automatically use new key versions from your Key Vault.
+
+- Changing the Disk Encryption Set URI (`diskEncryptionSetID`) in the Palette eXtended Kubernetes pack configuration in
+  Palette will trigger a node repave.
+
+## Prerequisites
+
+- To create the Azure Key Vault and Disk Encryption Set, you will need an Azure user account with the **Contributor** or
+  **Owner** role on the resource group or subscription.
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:49]: We do not recommend the contributor role. We have gone to great lengths to document all the required Azure [permissions](https://deploy-preview-4558--docs-spectrocloud.netlify.app/clusters/public-cloud/azure/required-permissions/) so using the contributor is frowned upon. So if there are permission they need then let's call them out, but let's not tell people to use the overprivileged Contributor role because they will use it a point back to this doc.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 18]
```diff
@@ -0,0 +1,353 @@
...
+10. Click **Confirm updates**.
+
+11. Click **Save Changes**.
+
+#### Enable Disk Encryption on an Active Cluster
+
+Use the following steps to enable disk encryption on an active cluster using the
+[updated cluster profile](#enable-disk-encryption-on-a-cluster-profile).
+
+:::warning
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:226]: Good use of warning

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 19]
```diff
@@ -0,0 +1,353 @@
...
+
+1. Log in to [Palette](https://console.spectrocloud.com).
+
+2. Ensure you are in the correct project scope.
+
+3. From the left **Main Menu**, select **Profiles** and click the cluster profile that you want to edit.
+
+4. Create a new version of your cluster profile:
+
+   1. Click the version drop-down menu next to the cluster profile name, and click **Create new version**.
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:264]:
```suggestion
   1. Click the version **drop-down Menu** next to the cluster profile name, and click **Create new version**.
```

Drop-down Menu should be in bold as its a UI element.

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
@benradstone Great job explaining the workflow. The steps are clear and easy to follow 💪🏻 

Some layout changes are required, and I spotted other minor style issues. Once these are addressed, go and request a re-review.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 20]
```diff
@@ -0,0 +1,353 @@
...
+title: "Azure Disk Encryption"
+description:
+  "How to enable disk encryption on your Azure IaaS cluster using Disk Encryption Sets with customer-managed keys"
+icon: ""
+hide_table_of_contents: false
+tags: ["public cloud", "azure", "encryption", "security"]
+sidebar_position: 12
+---
+
+Palette supports disk encryption of your Azure Kubernetes cluster using Disk Encryption Sets with
```
[the remark this answers is not in the document]
Comment (benradstone) [reply docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:12]: I'll adjust the hyperlink to start from here, rather than just Customer-managed keys.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 21]
```diff
@@ -0,0 +1,353 @@
...
+[customer-managed keys](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption#customer-managed-keys).
```
Comment (benradstone) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:13]:
```suggestion
Palette supports disk encryption of your Azure Kubernetes cluster using [Disk Encryption Sets with
customer-managed keys](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption#customer-managed-keys).
```

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md (hunk 16 above)
Comment (benradstone) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:31]:
```suggestion
  <!-- prettier-ignore -->
- Azure Disk Encryption is only supported when using the
  <VersionedLink text="Palette eXtended Kubernetes (PXK)" url="/integrations/packs/?pack=kubernetes" /> pack.
```

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md (hunk 17 above)
Comment (benradstone) [reply docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:49]: They mention this in the Feature Overview, but I will break this down into permissions required for each task.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md (hunk 8 above)
Comment (benradstone) [reply docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:78]: Thank you, I was looking for an example like that.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 22]
```diff
@@ -0,0 +1,353 @@
...
+    1. Log in to the [Azure Portal](https://portal.azure.com/).
+
+    2. In the search bar, look for **Disk Encryption Sets**. Click on the service when found.
+
+    3. Find your Disk Encryption Set from the list and click on it to view details.
+
+    4. On the **Overview** page, click **JSON View** in the **Essentials** section. The Resource ID for the Disk
+       Encryption Set is displayed at the top.
+
+    5. Click the **Copy to clipboard** icon for the Resource ID and paste it into the `diskEncryptionSetID` field in the
```
[the remark this answers is not in the document]
Comment (benradstone) [reply docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:131]: I wrote it as an implication that the user will see their subscriptionId and resourceGroup in the full URI.

The Resource ID contains the full URI (including the Sub and RG IDs), so to avoid confusion, I'll remove the edged-brackets.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 23]
```diff
@@ -0,0 +1,353 @@
...
+       Palette YAML editor.
+
+    </details>
+
+    ```yaml
+    cloud:
+      azure:
+        diskEncryptionSetID: "/subscriptions/<subscriptionId>/resourceGroups/<resourceGroup>/providers/Microsoft.Compute/diskEncryptionSets/<diskEncryptionSet>"
```
Comment (benradstone) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:139]:
```suggestion
        diskEncryptionSetID: "/subscriptions/subscriptionId/resourceGroups/resourceGroup/providers/Microsoft.Compute/diskEncryptionSets/diskEncryptionSet"
```

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 24]
```diff
@@ -0,0 +1,353 @@
...
+   3. Click **Confirm**.
+
+5. Select the Kubernetes layer to view the **Edit Pack** drawer.
+
+   :::warning
+
+   Ensure that you are using
+
+   <VersionedLink text="Palette eXtended Kubernetes (PXK)" url="/integrations/packs/?pack=kubernetes" /> as the Azure
+   Disk Encryption option is only available to this pack.
```
Comment (benradstone) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:175]:
```suggestion
   <!-- prettier-ignore -->
   Ensure that you are using
   <VersionedLink text="Palette eXtended Kubernetes (PXK)" url="/integrations/packs/?pack=kubernetes" />, as the Azure
   Disk Encryption option is only available to this pack.
```

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 25]
```diff
@@ -0,0 +1,353 @@
...
+      Encryption Set is displayed at the top.
+   5. Click the **Copy to clipboard** icon for the Resource ID and paste it into the `diskEncryptionSetID` field in the
+      Palette YAML editor.
+
+   </details>
+
+   ```yaml
+   cloud:
+     azure:
+       diskEncryptionSetID: "/subscriptions/<subscriptionId>/resourceGroups/<resourceGroup>/providers/Microsoft.Compute/diskEncryptionSets/<diskEncryptionSet>"
```
Comment (benradstone) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:214]:
```suggestion
       diskEncryptionSetID: "/subscriptions/subscriptionId/resourceGroups/resourceGroup/providers/Microsoft.Compute/diskEncryptionSets/diskEncryptionSet"
```

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md (hunk 2 above)
Comment (benradstone) [reply docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:221]: I thought about doing this originally. I kept them separate to call out the admonition warning before a user started this part, but it makes sense to merge them together.

I'll call it out either at the beginning, or before they move onto these steps instead.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md (hunk 7 above)
Comment (benradstone) [reply docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:325]:
```suggestion
6. Repeat steps 4 and 5 for each disk in your cluster.
```

Comment (benradstone) [reply docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:325]: @caroldelwing In our style-guide, we stipulate:

> In how-to steps, if instructions refer to a specific step or steps, use numbers instead of spelling out the word.

File: docs/docs-content/clusters/public-cloud/azure/create-azure-cluster.md [hunk 26]
```diff
@@ -67,6 +67,14 @@ to create an IaaS Kubernetes cluster in Azure that Palette manages.
...
+
+  <VersionedLink text="Palette eXtended Kubernetes (PXK)" url="/integrations/packs/?pack=kubernetes" /> pack, and have
+  the **Use Azure Disk Encryption** preset enabled.
+
+  Review [Azure Disk Encryption](./azure-disk-encryption.md) for guidance.
```
[the remark this answers is not in the document]
Comment (benradstone) [reply docs/docs-content/clusters/public-cloud/azure/create-azure-cluster.md:76]:
```suggestion
<!-- prettier-ignore -->
- If you want to enable Azure Disk Encryption on your cluster, ensure you have created a Key Vault and Disk Encryption Set. Your cluster profile must also be configured to use the <VersionedLink text="Palette eXtended Kubernetes (PXK)" url="/integrations/packs/?pack=kubernetes" /> pack, and have the **Use Azure Disk Encryption** preset enabled.

  Review [Azure Disk Encryption](./azure-disk-encryption.md) for guidance.
```

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 27]
```diff
@@ -0,0 +1,353 @@
...
+- To create the Azure Key Vault and Disk Encryption Set, you will need an Azure user account with the **Contributor** or
+  **Owner** role on the resource group or subscription.
+
+- Create an
+  [Azure Key Vault](https://learn.microsoft.com/en-us/azure/virtual-machines/disks-enable-customer-managed-keys-portal#set-up-your-azure-key-vault)
+  with the following necessary configuration:
+
+  - Resource access:
+
+    - :white_check_mark: **Azure Virtual Machines for deployment**
```
[the remark this answers is not in the document]
Comment (benradstone) [reply docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:56]: No, I should have read our Emoticons section better in our style guide.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md (hunk 13 above)
Comment (benradstone) [reply docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:173]: We do call it out in limitations. Not sure if this suffices or not.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 28]
```diff
@@ -0,0 +1,383 @@
...
+
+- Changing the Disk Encryption Set URI (`diskEncryptionSetID`) in the Palette eXtended Kubernetes pack configuration in
+  Palette will trigger a node repave.
+
+## New Cluster Profile
+
+### Prerequisites {#prerequisites-new-cluster-profile}
+
+- An Azure user account with the following roles to create the Azure Key Vault and Disk Encryption Set with the
+  necessary Key Vault access policies.
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:48]: Might be good to link to the Azure docs explanation of the built in key roles
https://learn.microsoft.com/en-us/azure/key-vault/general/rbac-guide?tabs=azure-cli#azure-built-in-roles-for-key-vault-data-plane-operations

**Review by karl-cardenas-coding — APPROVED** [review]
Great changes @benradstone . The only thing I noticed is that there is a bit of [parentheses usage](https://spectrocloud.atlassian.net/wiki/spaces/DE/pages/1765933057/Spectro+Cloud+Internal+Style+Guide). We avoid that style of writing as we consider it distracting.

**Comment by benradstone** [comment]
> Great changes @benradstone . The only thing I noticed is that there is a bit of [parentheses usage](https://spectrocloud.atlassian.net/wiki/spaces/DE/pages/1765933057/Spectro+Cloud+Internal+Style+Guide). We avoid that style of writing as we consider it distracting.

I've removed some usage of parentheses now.

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 29]
```diff
@@ -0,0 +1,386 @@
...
+
+  | Task                                                                          | Required Role                            |
+  | ----------------------------------------------------------------------------- | ---------------------------------------- |
+  | Create Key Vault and Key                                                      | Key Vault Contributor                    |
+  | Create Disk Encryption Set & Assign Key Vault Key                             | Key Vault Administrator                  |
+  | Assign Key Vault Access Policies (GET, WRAP KEY, UNWRAP KEY)                  | Key Vault Crypto Service Encryption User |
+  | (Optional) Assign User-Assigned Identity to Key Vault and Disk Encryption Set | Managed Identity Operator                |
+
+  For more information, see
```
[the remark this answers is not in the document]
Comment (benradstone) [reply docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:58]:
```suggestion
  For more information, visit
```

**Review by caroldelwing — APPROVED** [review]
Everything looks great! Congratulations, Ben! :rocket:

File: docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md [hunk 30]
```diff
@@ -0,0 +1,388 @@
...
+[Disk Encryption Sets with customer-managed keys](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption#customer-managed-keys).
+
+By default, Azure encrypts all managed disks with
+[platform-managed keys](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption#platform-managed-keys),
+however, customer-managed keys enable you to have greater control over your key management.
+
+<!-- prettier-ignore -->
+You can use Disk Encryption Sets to encrypt your nodes' Operating System and data disks by selecting a preset in the
+
+<VersionedLink text="Palette eXtended Kubernetes" url="/integrations/packs/?pack=kubernetes" /> pack.
```
Comment (benradstone) [inline docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md:22]:
```suggestion
<!-- prettier-ignore -->
You can use Disk Encryption Sets to encrypt your nodes' Operating System and data disks by selecting a preset in the <VersionedLink text="Palette eXtended Kubernetes" url="/integrations/packs/?pack=kubernetes" /> pack.
```
